WonderMake/Utilities/Container/ContainerTraits.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <type_traits>
+
+class BaseRequirement {};
+
+template<typename T>
+concept ContainerTrait = std::is_base_of_v<BaseRequirement, T>;
+
+template<typename T, typename... Ts>
+concept ContainsTrait = std::disjunction_v<std::is_same<T, Ts>...>;
+
+class Iterable : public BaseRequirement {};
+class Indexable : public BaseRequirement {};
+class RandomAccess : public BaseRequirement {};
+class StableElements : public BaseRequirement {};
+class ConstantInsertion : public BaseRequirement {};
+class ConstantDeletion : public BaseRequirement {};
```

Comment:
Allows extension outside of the engine, we can have inheritance to allow other operations(if we add a Sort requirement maybe it could inherit from Iterable and Indexable for instance).